src/reducers/author.js

```diff
@@ -31,9 +34,25 @@ export const author = (state = initialStates, action = {}) => {
         isFetching: true
       })
     case types.FETCH_AUTHOR_COLLECTION_SUCCESS:
-      let actionObjToAdd = _.omit(action, [ 'type', 'authorId' ])
-      actionObjToAdd.isFetching = false
-      return _.mergeWith({}, state, actionObjToAdd, concatIfIsArray)
+      const {
+        authorId,
+        response,
+        currentPage,
+        totalResults,
+        totalPages,
+        receivedAt } = action
+      const saveToState = {
+        authorId,
+        isFetching: false,
+        [authorId]: {
+          collectIndexList: _.get(response, 'result', []),
+          currentPage,
+          totalResults,
+          receivedAt,
+          isFinish: (currentPage - REQUEST_PAGE_START_FROM + 1 >= totalPages)
+        }
+      }
+      return _.mergeWith({}, state, saveToState, concatCollectIndexList)
```

Comment:
The difference between using `mergeWith` and only concacting the `collectIndexList` is that  using `mergeWith` will keep the author(A) collection list and pagination info in state when user go to other author(B)'s page. It will not need to refetch collections list of authorA if it has been saved in state.